src/composables/usePermalink.ts

```diff
@@ -0,0 +1,185 @@
+import { ref, type Ref } from 'vue'
+import type { Map } from 'ol'
+import type { View } from 'ol'
+
+export interface PermalinkState {
+  zoom: number
+  center: [number, number]
+  currentMgrsTileId: string | null
+  activeTileId: string | null
+  secondActiveTileId: string | null
+}
+
+export function usePermalink() {
```

Comment:
The functionality in this file is not a good candidate for a composable. There is no module level state, and the only composable level state (`shouldUpdate`) does not need to be a ref, because it is not used in a component.

I'd suggest moving this file to the `functions/` folder, and exporting the functions separately. For the `updatePermalink()` function, `shouldUpdate` can be a plain boolean variable, defined in the scope of the `setupPermalink()` function.